### ISSUE TITLE
Notes on using CFT for First Time

### ISSUE THREAD

**Issue by ghost** [F5Networks/f5-aws-cloudformation#31]
Hey guys.  Thanks for creating these templates and documentation.  They are extremely helpful!

FWIW, here are a couple of details that tripped me up when using these from scratch for the first time.  I think some tweaks to the prompts/docs could prevent others for having the same issues.

(I was using the 1 NIC/Good/1 GB Hourly template - literally the first yellow button on the template page. I'm not sure if these issues occur with other templates, but it seems like they might.)

1.) The first time through, after a long confusing delay, creating the stack failed because I needed to subscribe to the specific VE I wanted in the AWS Marketplace.  If would have been nice if that info was in the hourly stack descriptions (maybe with a link to facilitate subscribing).

2.) Obviously, F5 can't control the confusing "Region" selecting issue with CFTs, but maybe you can add a big message on the second page  that says "Make sure you've selected the proper region before continuing."  I'm guessing many VEs are accidentally created in Virginia by mistake.

3.) "Source Address(es) for Management Access" confused me for a while.  Specifically the description line that ended with "instances" (plural).  That caused me to think about the instances (plural) behind the load balancer, not the instance that contains the load balancer.  The description in the docs weren't too helpful either.  I know this is wordsmithing, but maybe something like: "IP Address(es) for Management Access: _______  - For SSH and web management access. Often this is your public IP address (/32)."

4.) "Source Address(es) for Web Application Access (80/443)" is still somewhat confusing to me.  Should it be "0.0.0.0/0" if the load balancer is fronting a public website?  I see that these settings are used in the security group that gets setup.  Maybe make that more clear in the description (i.e., mention that fact above both of these fields in the template)?

Looking back now, I think the term "Source" confused me.

5.) After the template runs, it is very unclear what to do next.  The docs state "If you want access to the BIG-IP web-based Configuration utility, you must first SSH into the BIG-IP VE using the SSH key you provided in the template. You can then create a user account with admin-level permissions on the BIG-IP VE to allow access if necessary."

The problem is that I don't setup BIG-IP LTMs from scratch every day (which is possibly my main message here in this post).  "Create a user account with admin-level permissions" needs to be clearly defined.   Eventually, I found a DevCentral article that showed someone logging in with ssh to the "admin" account (good to know) and entering "modify auth admin password" to change the admin password (also good to know).

6.) Something else that had slipped my mind was the default address for the configuration utility.  http://xx.xx.xx.xx didn't work.  https://xx.xx.xx.xx didn't work either.  Looking at the security group for the BIG-IP made me think that one of those two should have worked.  I wasted a couple of hours in frustration until discovering/remembering that it was https://xx.xx.xx.xx:8443  (doh!)

Again, these are simple things that - for a harried administrator - can really cause a ton of frustration.  Adding super-clear steps to the setup instructions (especially at the end) would really help.

- Chip

**Comment by chipascc** [comment]
More feedback for the 1-nic configuration:

(FWIW, I'm using a pool of servers that expect traffic on port 8080.  So my virtual server moves traffic from port 80 to port 8080 and back.)

When I created my HTTP virtual server, I had to use the primary private IP address in order for things to work.  Using the secondary private IP address (like the instructions specified) didn't work - no in-bound traffic was getting to the virtual server.  Not sure why that was.

The diagram with the 1-nic CFT shows the VLAN as "internal".  One my box it was named "external."

I also had to enable SNAT automap for my Virtual Server in order for requests to get back out.
